addons/webhook/certificates.go

```diff
@@ -0,0 +1,114 @@
+package webhook
+
+import (
+	"bytes"
+	cryptorand "crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"math/big"
+	"time"
+)
+
+type TLSCredentials struct {
+	CAName       string
+	Organization string
+	Hostname     string
+	DnsNames     []string
+	ValidFrom    time.Time
+	MaxAge       time.Duration
+	CA_cert      *[]byte
```

Comment:
this should probably be private. (lowercase) best to create a function to provide the values if they are needed. Same as getPEM functions, something like getDER (the ca_cert, server_cert, and rsa keys are DER encoded). but it will be bes to force the caller to specify the encoding they want.